euphonic/structure_factor.py

```diff
@@ -31,43 +34,51 @@ class StructureFactor(object):
         Phonon frequencies per q-point and mode
     structure_factors : (n_qpts, 3*crystal.n_atoms) float Quantity
         Structure factor per q-point and mode
+    weights : (n_qpts,) float ndarray
+        The weight for each q-point
     temperature : float Quantity or None
         The temperature used to calculate any temperature-dependent
         parts of the structure factor (e.g. Debye-Waller, Bose
         population factor). None if no temperature-dependent effects
         have been applied
     """
-
-    def __init__(self, crystal, qpts, frequencies, structure_factors,
-                 temperature=None):
+    def __init__(self, crystal: Crystal, qpts: np.ndarray,
```

Comment:
I think this `__init__()` should be using `__super__()__init__(*some_args)` to avoid redundancy.